nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,49 +148,54 @@ def run_conversion(
             f"({len(image_series_kwargs_list)}) vs. file_paths ({len(self.source_data['file_paths'])})!"
         )
 
-        for j, file in enumerate(file_paths):
-            timestamps = starting_times[j] + get_movie_timestamps(movie_file=file)
-
-            if len(starting_times) != len(file_paths):
-                starting_times.append(timestamps[-1])
+        def _check_duplicates(image_series_kwargs_list):
+            image_series_kwargs_list_keys = [i["name"] for i in image_series_kwargs_list]
+            if len(set(image_series_kwargs_list_keys)) < len(image_series_kwargs_list_keys):
+                assert external_mode, "For multiple video files under the same ImageSeries name, use exernal_mode=True."
+            keys_set = []
+            image_series_kwargs_list_unique = []
+            for n, image_series_kwargs in enumerate(image_series_kwargs_list):
+                if image_series_kwargs["name"] not in keys_set:
+                    keys_set.append(image_series_kwargs["name"])
+                    image_series_kwargs_list_unique.append(dict(image_series_kwargs, data=[file_paths[n]]))
+                else:
+                    idx = keys_set.index(image_series_kwargs["name"])
+                    image_series_kwargs_list_unique[idx]["data"].append(file_paths[n])
+            return image_series_kwargs_list_unique
```

Comment:
The main goal of this, as I understand, is to handle multiple file path inputs to a single ImageSeries object via external mode.

It's misleading to fold in the `data` keyword to the `image_series_kwargs` in this manner, when the `data` field is actually nothing like that even when you use `external_mode=True`, which is what leads to the awkward popping in the iteration of `image_series_kwargs_list_updated` only to be properly re-added at the end of the method when `external_mode=False`.

It might be better design to simply use this like `file_list = _extract_file_list(image_series_kwargs_list)` which can then return the `file_list` without any popping, and can still assert uniqueness of names when `external_mode=False`. This would also make the code more readable as well.

Reply:
yeah okay, indeed its not very readable. See here https://github.com/catalystneuro/nwb-conversion-tools/pull/372/commits/62b39d288848d34462fe08dab9db7eb540f506cc

Reply:
yes I agree its not very readable : https://github.com/catalystneuro/nwb-conversion-tools/pull/372/commits/62b39d288848d34462fe08dab9db7eb540f506cc